crates/sui-node/src/main.rs

```diff
@@ -57,6 +69,12 @@ fn main() {
     );
     config.supported_protocol_versions = Some(SupportedProtocolVersions::SYSTEM_DEFAULT);
 
+    let run_with_range = match (args.run_with_range_epoch, args.run_with_range_checkpoint) {
+        (None, Some(checkpoint)) => RunWithRange::Checkpoint(checkpoint),
+        (Some(epoch), None) => RunWithRange::Epoch(epoch),
+        (None, None) | (Some(_), Some(_)) => RunWithRange::None,
```

Comment:
nit: `Some, Some` is not possible with the exclusive group, right? Can we instead do `unreachable!("This variant is not possible")` for that variant to be explicit?